src/block.rs

```diff
@@ -552,61 +552,93 @@ impl Block {
 }
 
 #[derive(Copy, Clone, Debug, PartialEq)]
-pub enum Comparator {
+pub enum Operator {
+    /// Can perhaps be replaced by wrapping the operator in an option.
     None,
-    /// Eq is also Assign
+    /// The = operator. Can be used as assignment, or to open a scope.
     Eq,
-    /// The == operator, which means Eq but cannot be used to assign
-    EEq,
-    /// The ?= operator
-    QEq,
-    Lt,
-    Gt,
-    Le,
-    Ge,
-    Ne,
+    /// The ?= operator Can be used to open a scope on the condition that it exists.
+    MaybeEq,
+    /// ==
+    Equals,
+    /// !=
+    NotEquals,
+    /// <
+    SmallerThan,
+    /// >
+    GreaterThan,
+    /// <=
+    AtMost,
+    /// >=
+    AtLeast,
 }
 
-impl Comparator {
+impl Operator {
     pub fn from_str(s: &str) -> Option<Self> {
         if s == "=" {
-            Some(Comparator::Eq)
+            Some(Operator::Eq)
         } else if s == "?=" {
-            Some(Comparator::QEq)
+            Some(Operator::MaybeEq)
         } else if s == "==" {
-            Some(Comparator::EEq)
+            Some(Operator::Equals)
         } else if s == "<" {
-            Some(Comparator::Lt)
+            Some(Operator::SmallerThan)
         } else if s == ">" {
-            Some(Comparator::Gt)
+            Some(Operator::GreaterThan)
         } else if s == "<=" {
-            Some(Comparator::Le)
+            Some(Operator::AtMost)
         } else if s == ">=" {
-            Some(Comparator::Ge)
+            Some(Operator::AtLeast)
         } else if s == "!=" {
-            Some(Comparator::Ne)
+            Some(Operator::NotEquals)
         } else {
             None
         }
     }
-
     pub fn from_token(token: &Token) -> Option<Self> {
         Self::from_str(token.as_str())
     }
+
+    /// Returns true if the operator is a comparison operator, or if it's an `=` operator,
+    /// which serves as an alias for the equality comparator.
+    pub fn is_comparator(self) -> bool {
+        matches!(
+            self,
+            Operator::Eq
+                | Operator::Equals
+                | Operator::NotEquals
+                | Operator::AtLeast
+                | Operator::AtMost
+                | Operator::GreaterThan
+                | Operator::SmallerThan
+        )
+    }
+    /// Substitutes `==` for `=`.
+    /// This is to allow `=` to be used as an alias for equality,
+    /// even though that's technically incorrect.
```

Comment:
Sorry but this is a very un-pdx-script attitude. = is the default equality operator, used *everywhere* in over 4 million lines of script, except for four places where == is used.

Please treat = like a normal operator.

Reply:
This does convert `=` to `==`, so both are handled the same. 

Basically, the `to_comparator()` function can be called when the operator is expected to be a comparator, and you'll know that it'll be an Equals, not an Eq. This way, you don't have to do two checks instead of one later-on in the code. 

Again, this is only in tiger's codebase. As far as pdx script is concerned, both are treated the same and no warnings are issued to the user.

Reply:
Okay but even within tiger's code base, = should be treated as the normal equality operator and == as the weird exception, otherwise you get an impedance mismatch with the code it's scrutinizing.

Examples should also use =.

Reply:
The comments are misleading here, but to my mind, the enum values don't actually have anything to do with notation, they should be viewed as different things. One represents equality comparison, the other assignment and opening scopes. I don't see Operator::Equals as having anything to do with `==` or `=`.

But I think the real problem is having both in the first place. I think we should merge them. Within paradox script, there clearly isn't a real distinction in how they're handled. In fact, `==` seems to be some kind of weird little-known, little-used alias for `=`. We should just serialise both to Comparator::Equals.